scripts/get-terraform-supported-versions.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+get_last_day_of_month() {
```

Comment:
nit: feel free to discard this, but I am wondering if we can avoid creating this type of well known code. I am not sure how to do it, but I wonder if it is possible to leverage an existing public reliable GH action step that returns "last day of month" 😃

Reply:
Totally agree! I've looked for both GH actions and bash tools for date manipulation and haven't found anything which can work both in locally and in the CI that makes it easier than just using this code here.

Reply:
what about running this the first day of the month: 0 0 1 * *
and test suite the rest: 0 0 2-31 * *